source/components/ui/msm-click-to-edit/msm-click-to-edit.directive.js

```diff
@@ -0,0 +1,126 @@
+/** Based on https://github.com/GabiGrin/angular-editable-text */
+(function () {
+  'use strict';
+
+  /**
+   * @ngdoc directive
+   * @name components.ui.msmEditableText
+   * @restrict 'E'
+   *
+   * @description Renders an editable text input
+   *
+   * @param {string} msmEditableText The model to be edited
+   * @param {string} editMode The edit mode
+   * @param {string} placeholder The placeholder for the model if empty
+   * @param {function} onChange A callback function on model change
+   */
+  angular.module('msm.components.ui')
+      .directive('msmEditableText', EditableText);
+
+  function EditableText() {
+    return {
+      scope: {
+        msmEditableText: '=',
+        editMode: '=',
+        placeholder: '@',
+        onChange: '&'
+      },
+      transclude: true,
+      templateUrl: 'components/ui/msm-click-to-edit/msm-click-to-edit.html',
+      link: function (scope, elem, attrs) {
+        var input = elem.find('input');
+        var lastValue;
+        scope.saveValue = true;
+        scope.isEditing = !!scope.editMode;
+        scope.editingValue = scope.msmEditableText;
+
+        scope.onBlur = function() {
+          scope.isEditing = false;
+        };
+
+        scope.onKeyPress = function(event) {
+          if (event) {
+            // enter
```

Comment:
Thanks for comment so I know what `13` means. Even better would be a const  variable:
```javascript
if (event.which === ENTER) {
```